pkg/util/git/git.go

```diff
@@ -64,7 +65,14 @@ func (gr *GitCLIRepository) Clone(ctx context.Context, options CloneOptions) err
 		}
 
 		args = append(args, options.Args...)
-		out, err := command.CombinedOutput(ctx, gr.LocalPath, expand.ListEnviron(os.Environ()...), "git", args...)
+		os.Setenv("GIT_SSH_COMMAND", "ssh -oBatchMode=yes")
+		gitEnv := []string{
+			"GIT_TERMINAL_PROMPT=0",
+			"GIT_SSH_COMMAND='ssh -oBatchMode=yes'",
+		}
+		gitEnv = append(gitEnv, os.Environ()...)
+		out, err := command.CombinedOutput(ctx, gr.LocalPath, expand.ListEnviron(gitEnv...), "git", args...)
+		os.Unsetenv("GIT_SSH_COMMAND")
```

Comment:
Is `os.Setenv()` and `os.Unsetenv()` required, or is passing `gitEnv` when running the command sufficient?

Reply:
It's required. `GIT_SSH_COMMAND ` will be used by the underlying ssh command.

Reply:
Yes, it is, I seem to misunderstand it.